tensorflow/compiler/mlir/tools/kernel_gen/transforms/tf_to_jit_invocations.cc

```diff
@@ -211,51 +222,121 @@ struct PackJITCompileOpPattern
+    auto shape = rewriter.create<shape::ShapeOfOp>(loc, arg);
+    auto num_elems = rewriter.create<shape::NumElementsOp>(loc, shape);
+    Value coniditon_check_main = rewriter.create<arith::CmpIOp>(
+                    loc, arith::CmpIPredicate::sgt, num_elems, shape_size_limit);
+
+    Value conditional_path =
+                rewriter.create<scf::IfOp>(
+                  loc, op->getResultTypes(), coniditon_check_main,
+                  [&](OpBuilder &b, Location l) {
+                    auto jit_compile_op = rewriter.create<tf_framework::JITCompileOp>(
+                      loc, rewriter.getType<tf_framework::JITCallableType>(), llvm::None);
+                      BlockAndValueMapping bvm;
+                      {
+                        OpBuilder::InsertionGuard guard(rewriter);
+                        Block *block =
+                            rewriter.createBlock(&jit_compile_op.body(), {}, operand_types);
+                        for (auto it : llvm::zip(op->getOperands(), block->getArguments()))
+                          bvm.map(std::get<0>(it), std::get<1>(it));
+                        rewriter.setInsertionPointToStart(block);
+                        rewriter.clone(*op, bvm);
+                        auto new_op = rewriter.clone(*op, bvm);
+                        rewriter.create<tf_framework::JITCompileYieldOp>(loc, TypeRange{},
+                                                                        new_op->getResults());
+                      }
+                    auto jit_execute_op = rewriter.create<tf_framework::JITExecuteOp>(
+                        loc, result_types, Value(), jit_compile_op.result(), op->getOperands());
+                          b.create<scf::YieldOp>(l, jit_execute_op.results());
+                     },
+                     [&](OpBuilder &b, Location l) {
+                       auto new_op = rewriter.clone(*op);
+                       b.create<scf::YieldOp>(l, new_op->getResult(0));
+                     })
+                    .getResult(0);
+
+  rewriter.replaceOp(op, conditional_path);
+  return success();
+  }
+};
 }  // namespace
 
-void PopulateTFToJITInvocationPatterns(MLIRContext *ctx,
-                                       RewritePatternSet *patterns,
+void PopulateTFToJITInvocationPatterns(MLIRContext* ctx,
+                                       RewritePatternSet* patterns,
                                        llvm::ArrayRef<int64_t> tile_sizes,
                                        llvm::ArrayRef<int64_t> unroll_factors,
                                        int64_t max_supported_rank,
-                                       bool enable_ftz, bool cpu_codegen) {
-  patterns->insert<TFToJITInvocationsPattern>(ctx);
+                                       bool enable_ftz, bool cpu_codegen,
+                                       bool jit_i64_indexed_for_large_tensors) {
+  if (jit_i64_indexed_for_large_tensors) {
+    patterns->insert<TFT64BitIndexerPattern>(ctx);
```

Comment:
Can we have a more descriptive name for `TFT64BitIndexerPattern`? E.g. `TFToI64JITInvocationForLargeTensorsPattern`

Reply:
name changed.